src/java.base/share/classes/java/nio/Direct-X-Buffer.java.template

```diff
@@ -75,15 +76,32 @@ class Direct$Type$Buffer$RW$$BO$
 
 #if[byte]
 
-    private record Deallocator(long address, long size, int capacity) implements Runnable {
-        private Deallocator {
+    private static final class Deallocator implements Runnable {
+
+        private static final AtomicIntegerFieldUpdater<Deallocator> UPDATER =
```

Comment:
It would be possible to use `Unsafe` directly instead of AIFU.

Reply:
+1, just use `Unsafe` directly here. `AIFU` would probably even worse than `AtomicInteger` if we want to minimize dependencies. We already use Unsafe in adjacent classes here.